internal/trace/listener.go

```diff
@@ -149,7 +149,9 @@ func startFunctionExecutionSpan(ctx context.Context, mergeXrayTraces bool, isDdS
 		span.SetTag("_dd.parent_source", "xray")
 	}
 
-	return span
+	ctx = context.WithValue(ctx, extension.DdSpanId, fmt.Sprint(span.Context().SpanID()))
```

Comment:
as a sanity check, that SpanID gets replaced here https://github.com/DataDog/datadog-agent/blob/570ed026d8dc3afc4324610db8f922027f8786ca/pkg/serverless/invocationlifecycle/trace.go#L110-L111?

Reply:
Exactly right. You'll notice that the tracer _and_ the extension are creating `aws.lambda` spans. However, the extension knows to throw out the one made in the tracer. The extension will ensure that both of these `aws.lambda` spans have the same trace context (span id, trace, id, parent id).

We do it this way because we want to make sure the tracer has access to the `aws.lambda` span as the parent of any spans it creates itself.